internal/tracing/manifests/otelcol/exporter.go

```diff
@@ -0,0 +1,103 @@
+package otelcol
+
+import (
+	"fmt"
+
+	"github.com/ViaQ/logerr/v2/kverrors"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func ConfigureExportersSecrets(cfg *map[string]interface{}, secret corev1.Secret, annotation string) error {
+	otelExporterName, ok := secret.Annotations[annotation]
+	if !ok {
+		return nil
+	}
+
+	exporters, err := getExporters(cfg)
+	if err != nil {
+		return err
+	}
+
+	for name, config := range *exporters {
+		exporterName := name.(string)
+		if otelExporterName != exporterName {
+			continue
+		}
+		var configMap map[interface{}]interface{}
+		if config == nil {
+			configMap = make(map[interface{}]interface{})
+			(*exporters)[otelExporterName] = configMap
+		} else {
+			configMap = config.(map[interface{}]interface{})
+		}
+
+		configureExporterSecrets(&configMap, secret)
+
+	}
+	return nil
+}
+
+func ConfigureExportersEndpoints(cfg *map[string]interface{}, cm corev1.ConfigMap, annotation string) error {
```

Comment:
Nit, but is there a reason the pass a pointer to this map? Since the map is always already always a reference.